driver-sync/src/test/functional/com/mongodb/client/AbstractUnifiedTest.java

```diff
@@ -305,12 +318,20 @@ public void shouldPassAllOutcomes() {
             List<CommandEvent> expectedEvents = getExpectedEvents(definition.getArray("expectations"), databaseName, null);
             List<CommandEvent> events = commandListener.getCommandStartedEvents();
 
-            assertEventsEquality(expectedEvents, events, lsidMap);
+            assertEventsEquality(expectedEvents, events.subList(0, expectedEvents.size()), lsidMap);
         }
 
         BsonDocument expectedOutcome = definition.getDocument("outcome", new BsonDocument());
         if (expectedOutcome.containsKey("collection")) {
-            List<BsonDocument> collectionData = collectionHelper.find(new BsonDocumentCodec());
+            BsonDocument collectionDocument = expectedOutcome.getDocument("collection");
```

Comment:
The new write concern tests contain aggregates containing `$out` or `$merge` which writes results to another collection. If the collection document contains a `name` field, we need to check the collection specified by `name` for the results.